luigi/contrib/sge.py

```diff
@@ -0,0 +1,335 @@
+        if self.__module__ == '__main__':
+            d = pickle.dumps(self)
+            module_name = os.path.basename(sys.argv[0]).rsplit('.', 1)[0]
+            d = d.replace('(c__main__', "(c" + module_name)
+            open(self.job_file, "w").write(d)
+
+        else:
+            pickle.dump(self, open(self.job_file, "w"))
+
+    def _run_job(self):
+
+        # Build a qsub argument that will run sge_runner.py on the directory we've specified
+        runner_path = sge_runner.__file__
+        if runner_path.endswith("pyc"):
+            runner_path = runner_path[:-3] + "py"
+        job_str = 'python {0} "{1}"'.format(runner_path, self.tmp_dir)  # enclose tmp_dir in quotes to protect from special escape chars
+
+        # Build qsub submit command
+        self.outfile = os.path.join(self.tmp_dir, 'job.out')
+        self.errfile = os.path.join(self.tmp_dir, 'job.err')
+        submit_cmd = build_qsub_command(job_str, self.task_family, self.outfile,
+                                        self.errfile, self.parallel_env, self.n_cpu)
+        logger.debug('qsub command: \n' + submit_cmd)
+
+        # Submit the job and grab job ID
+        output = subprocess.check_output(submit_cmd, shell=True)
+        self.job_id = parse_qsub_job_id(output)
+        logger.debug("Submitted job to qsub with response:\n" + output)
+
+        self._track_job()
+
+        # Now delete the temporaries, if they're there.
+        if self.tmp_dir and os.path.exists(self.tmp_dir):
+            logger.info('Removing temporary directory %s' % self.tmp_dir)
+            shutil.rmtree(self.tmp_dir)
+
+    def _track_job(self):
+        while True:
+            # Sleep for a little bit
+            time.sleep(POLL_TIME)
+
+            # See what the job's up to
+            # ASSUMPTION
+            qstat_out = subprocess.check_output(['qstat'])
+            sge_status = parse_qstat_state(qstat_out, self.job_id)
+            if sge_status == 'r':
+                logger.info('Job is running...')
+            elif sge_status == 'qw':
+                logger.info('Job is pending...')
+            elif 'E' in sge_status:
+                logger.error('Job has FAILED:\n' + '\n'.join(self.fetch_task_failures()))
+                break
+            elif sge_status == 't' or sge_status == 'u':
+                # Then the job could either be failed or done.
+                errors = self.fetch_task_failures()
+                if not errors:
+                    logger.info('Job is done')
+                else:
+                    logger.error('Job has FAILED:\n' + '\n'.join(errors))
+                break
+            else:
+                logger.info('Job status is UNKNOWN!')
+                logger.info('Status is : %s' % sge_status)
+                raise Exception("job status isn't one of ['r', 'qw', 'E*', 't', 'u']: %s" % sge_status)
+
+
+class LocalSGEJobTask(SGEJobTask):
+    """A local version of SGEJobTask, for easier debugging."""
```

Comment:
I don't know much about SGE, but does this deserve more docs to explain how to run?